src/Mailer/Email.php

```diff
@@ -2084,11 +2085,23 @@ protected function _logDelivery($contents)
         }
         Log::write(
             $config['level'],
-            PHP_EOL . $contents['headers'] . PHP_EOL . PHP_EOL . $contents['message'],
+            PHP_EOL . $this->flatten($contents['headers']) . PHP_EOL . PHP_EOL . $this->flatten($contents['message']),
             $config['scope']
         );
     }
+    
+    
+    /**
+     * Converts $array to string if array given
+     * @param null $key
+     * @return string
+     */
+    protected function flatten($array = null)
```

Comment:
This must not be nullable.
Also, the param annotation is still invalid (null), must be array|string.